src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ProcessLinkerXmlBase.cs

```diff
@@ -92,18 +96,18 @@ protected virtual void ProcessXml(bool stripResource, bool ignoreResource)
 
                 // For embedded XML, allow not specifying the assembly explicitly in XML.
                 if (_resource != null)
-                    ProcessAssembly(_resource.Value.Assembly, nav, warnOnUnresolvedTypes: true);
+                    ProcessAssembly(_resource.Value.Module, nav, warnOnUnresolvedTypes: true);
 
             }
-            catch (Exception ex) when (!(ex is LinkerFatalErrorException))
+            catch (Exception ex)
             {
-                throw new LinkerFatalErrorException(MessageContainer.CreateErrorMessage(null, DiagnosticId.ErrorProcessingXmlLocation, _xmlDocumentLocation), ex);
+                throw ex;
```

Comment:
Could you leave what was here commented out? We'll probably want to revisit because telling the user about what XML we were processing sounds useful.